array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-unused-vars */
+function zip(first, second) {
+  var newArray = [];
+  for (var i = 0; i < first.length; i++) {
+    for (var s = 0; s < second.length; s++) {
+      if (i === s) {
+        var coupleArray = [];
+        coupleArray.push(first[i]);
+        coupleArray.push(second[s]);
+        newArray.push(coupleArray);
+      }
+    }
+  }
```

Comment:
Doing a nested loop like this is really inefficient from a computation standpoint. Try this:
- decide which array is shorter
- loop through the shorter array
- use the index in your loop on both arrays.